xmodule/capa/capa_problem.py

```diff
@@ -495,6 +509,51 @@ def get_grade_from_current_answers(self, student_answers):
 
         return newcmap
 
+    def get_grade_from_answers(
+        self, student_answers: Optional[dict], correct_map: Optional[CorrectMap] = None
+    ) -> CorrectMap:
+        """
+        This method is based on `get_grade_from_current_answers` but it is used when
+        `ENABLE_GRADING_METHOD_IN_PROBLEMS` feature flag is enabled.
+
+        This method optionally receives a `correct_map` to be used instead
```

Comment:
There is a subtlety in this new feature that the current comment doesn't capture well. Namely, with the new feature we need to be able to recalculate a student's score for this problem every time the grading policy changes, as well as every time the student adds a new answer. It follows that the problem's state information needs to be rich enough to support both cases. 

With the new state fields added for this feature, every time a student answers a new answer to the problem a snapshot of the current answer key is retained. Put another way, the applicable grade on the day the problem is answered is grandfathered in, and a subsequent change to the answer key won't change that grade: the overall grade may change with changes to the answer key, or with changes to the grading policy, but the grade assigned to an answer submitted today should remain today's grade.

Reply:
To avoid code repetition I opted to remove the `get_grade_from_answers` method, and instead updated the current `get_grade_from_current_answers` method to accept an optional parameter that receives a **correct_map**. Likewise, I added a new conditional to check if the feature is active, in which case the **correct_map** and **student_answers** passed as argument to the method will be used, and not those of the instance.

Reply:
That looks better; thank you! However, we still need the comment explaining the nuances of the implementation; could you add that? Thanks!

Reply:
Done, I added in the docstring the behavior when the grading method is enabled. I think this is clearer, any additional suggestions?